express/component_spec.py

```diff
@@ -0,0 +1,346 @@
+        """
+        Function that returns the specification of the kubeflow component
+        Returns:
+            The Kubeflow specification as a dictionary
+        """
+        if not all(
+            [
+                self.name,
+                self.description,
+                self.inputs,
+                self.outputs,
+                self.image,
+                self.command,
+            ]
+        ):
+            raise ValueError("Missing required attributes to construct specification")
+
+        specification = {
+            "name": self.name,
+            "description": self.description,
+            "inputs": [asdict(input_obj) for input_obj in self.inputs],
+            "outputs": [asdict(output_obj) for output_obj in self.outputs],
+            "implementation": {
+                "container": {"image": self.image, "command": self.command}
+            },
+        }
+
+        return specification
+
+
+class ComponentSubset:
+    """
+    Class representing an Express Component subset.
+    """
+
+    def __init__(self, specification: dict) -> None:
+        """
+        Initialize subsets
+        Args:
+            specification: the part of the component json representing the subset
+        """
+        self._specification = specification
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}({self._specification!r}"
+
+    @property
+    def fields(self) -> t.Mapping[str, Field]:
+        return types.MappingProxyType(
+            {
+                name: Field(name=name, type=field)
+                for name, field in self._specification["fields"].items()
+            }
+        )
+
+
+class ExpressComponent:
+    """
+    Class representing an Express component
+    Args:
+        yaml_spec_path: The yaml file containing the component specification
+    """
+
+    def __init__(self, yaml_spec_path: str):
+        self.yaml_spec = load_yaml(yaml_spec_path)
+        self._validate_spec()
+        self._kubeflow_component_specification = (
+            self.set_kubeflow_component_specification()
+        )
+        self._express_component_specification = (
+            self.set_express_component_specification()
+        )
+
+    def _validate_spec(self) -> None:
+        """Validate a component specification against the component schema
+        Raises: InvalidManifest when the manifest is not valid.
+        """
+        spec_schema = json.loads(
+            pkgutil.get_data("express", "schemas/component_spec.json")
+        )
+
+        base_uri = (Path(__file__).parent / "schemas").as_uri()
+        resolver = RefResolver(base_uri=f"{base_uri}/", referrer=spec_schema)
+        validator = Draft4Validator(spec_schema, resolver=resolver)
+
+        try:
+            validator.validate(self.yaml_spec)
+        except jsonschema.exceptions.ValidationError as e:
+            raise InvalidComponentSpec.create_from(e)
+
+    def set_kubeflow_component_specification(self) -> t.Dict[str, any]:
```

Comment:
Don't like the `set_` here, as the method is not setting anything. Same for the method below.